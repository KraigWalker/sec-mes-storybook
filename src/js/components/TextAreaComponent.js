import React from 'react';
import { Link } from 'react-router-dom';

class TextAreaComponent extends React.Component{
    constructor(props) {
        super(props);
        this.textChange = this.textChange.bind(this);
        this.sendData = this.sendData.bind(this);
        this.state = {
            messageText : '',
        }
    }
    textChange(e) {
        // console.log('text changes',e.target.value);
        this.setState({messageText : e.target.value});
    }
    sendData(){
        this.props.sendData(this.state.messageText);
    }
    render() {
        return (<div>
            <input type = 'text' name = 'txtbox'onChange = {this.textChange}/><br/><br/>
            <Link to='/securemessages'>
            <input type = 'button' name = 'cancel' value = 'Back'/>
            </Link>
            <input type = 'button' name = 'Save Draft' value = 'Save Draft'/>
            <input type = 'button' name = 'Send' value = 'Send' onClick ={this.sendData}/>
        </div>);
    }
}
export default TextAreaComponent;