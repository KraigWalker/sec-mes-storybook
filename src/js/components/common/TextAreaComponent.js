import React from 'react';
import { Link } from 'react-router-dom';
import autosize from 'autosize';

class TextAreaComponent extends React.Component {
    constructor(props) {
        super(props);
        this.textChange = this.textChange.bind(this);
        this.check = this.check.bind(this);
        this.state = {
            textLength: 0,
        }
    }
    componentDidMount(){
        if(this.props.isFromDraft) {
            this.refs.txtarea.value = this.props.draftData;
            this.props.textData(this.props.draftData);
        } else this.refs.txtarea.value = '';
    }
    textChange(e) {
        autosize(document.getElementById('message'));
        this.props.textData(e.target.value);
    }
    check(e) {
        // this.props.pastedData(e.clipboardData.getData('Text'));
    }
    render() {
        return (
            <textarea className="c-field__input c-field__input--txtarea"
                name="message"
                id="message"
                rows="10"
                cols="20"
                onChange={this.textChange}
                onPaste={this.check}
                value={this.props.messageBody}
                disabled={this.props.disableText}
                maxLength='43'
                ref="txtarea">
            </textarea>
        );
    }
}
export default TextAreaComponent;