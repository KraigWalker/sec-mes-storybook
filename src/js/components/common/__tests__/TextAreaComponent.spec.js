import React from 'react';
import { shallow, mount } from 'enzyme';
import TextAreaComponent from '../TextAreaComponent';

describe('TextAreaComponent Component Check', () => {
  const componentWrap = mount(<TextAreaComponent messageBody='text' disableText={true}/>);

  it("TextAreaComponent component should mount", () =>{
    expect(typeof componentWrap).toBe('object');
  });
  it("Props Check", () =>{
    expect(componentWrap.props().messageBody).toEqual('text');
    expect(componentWrap.props().disableText).toEqual(true);
    
  });
  it("Textarea Check", () =>{
    expect(componentWrap.find('textarea')).toHaveLength(1);
  });

  it('should render correctly', () => {
   const tree = shallow(
     <TextAreaComponent name='message' />
   );
   expect(tree).toMatchSnapshot();
 });

});