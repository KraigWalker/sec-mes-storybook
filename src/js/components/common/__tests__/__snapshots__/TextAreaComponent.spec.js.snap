// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAreaComponent Component Check should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextAreaComponent
    ariaId=""
    name="message"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-describedby": "",
      "aria-labelledby": "undefined",
      "className": "c-field__input c-field__input--txtarea",
      "cols": "20",
      "disabled": undefined,
      "id": "message",
      "maxLength": "infinity",
      "name": "message",
      "onChange": [Function],
      "onPaste": undefined,
      "rows": "10",
      "value": undefined,
    },
    "ref": [Function],
    "rendered": null,
    "type": "textarea",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-describedby": "",
        "aria-labelledby": "undefined",
        "className": "c-field__input c-field__input--txtarea",
        "cols": "20",
        "disabled": undefined,
        "id": "message",
        "maxLength": "infinity",
        "name": "message",
        "onChange": [Function],
        "onPaste": undefined,
        "rows": "10",
        "value": undefined,
      },
      "ref": [Function],
      "rendered": null,
      "type": "textarea",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
